src/qt/importdialog.cpp

```diff
@@ -0,0 +1,302 @@
+#include <qt/importdialog.h>
+#include <qt/forms/ui_importdialog.h>
+
+#include <qt/guiconstants.h>
+#include <qt/guiutil.h>
+#include <key_io.h>
+#include <qt/walletmodel.h>
+#include <interfaces/node.h>
+#include <wallet/wallet.h>
+#include <QKeyEvent>
+#include <QMessageBox>
+#include <QPushButton>
+#include <iostream>
+#include <QScrollBar>
+#include <tinyformat.h>
+#include <chain.h>
+
+ImportDialog::ImportDialog(Page _page, QWidget *parent, WalletModel *walletModel) :
+    QDialog(parent, GUIUtil::dialog_flags),
+    ui(new Ui::ImportDialog),
+    walletModel(walletModel),
+    page(_page)
+{
+    ui->setupUi(this);
+
+    switch (page) {
+        case importPubkey: {
+            ui->QStackContainer->setCurrentWidget(ui->importKeyPage);
+            this->setWindowTitle("Import Public Key");
+            ui->keyLabelIKP->setText("Public Key");
+            ui->keyEditIKP->setPlaceholderText("Enter a Public Key (e.g. 1NS17iag9jJgTHDlVXjvLCEnZuQ3rJDE9L)");
+            ui->rescanButtonIKP->setCheckState(Qt::Checked);
+            break;
+        } case importPrivkey: {
+            ui->QStackContainer->setCurrentWidget(ui->importKeyPage);
+            this->setWindowTitle("Import Private Key");
+            ui->keyLabelIKP->setText("Private Key");
+            ui->keyEditIKP->setPlaceholderText("Enter a Private Key (e.g. L4sTHokqoFZpcmJgsAG2yxrf2f71v6PYpK2qEkCvc2Fp8VVdMiU3)");
+            ui->rescanButtonIKP->setCheckState(Qt::Checked);
+            break;
+        } case importAddress: {
+            ui->QStackContainer->setCurrentWidget(ui->importAddress);
+            this->setWindowTitle("Import Address");
+            ui->rescanButtonIA->setCheckState(Qt::Checked);
+            ui->p2shButtonIA->setCheckState(Qt::Unchecked);
+            break;
+        }
+    }
+
+    GUIUtil::handleCloseWindowShortcut(this);
+}
+
+ImportDialog::~ImportDialog()
+{
+    delete ui;
+}
+
+void ImportDialog::on_Accept_clicked() {
+    switch (page) {
+        case importPubkey: {
+            std::string strKey = "";
+            strKey.assign(ui->keyEditIKP->text().toStdString().c_str());
+
+            if (strKey.empty()) {
+                QMessageBox::warning(this, tr("Error"), tr("Key is empty"));
+                break;
+            }
+
+            std::string strLabel;
+            strLabel.assign(ui->labelEditIKP->text().toStdString().c_str());
```

Comment:
In 3a75af09dbb6c182a86ce71b15fe0824d3c68ac5 "qt: implemented importpubkey, importprivkey, and importaddress GUI's"

This can be simplified to one line, here and elsewhere.

```suggestion
            std::string strLabel = ui->labelEditIKP->text().toStdString();
```

Reply:
done